dimitriosv/week2/index.html

```diff
@@ -0,0 +1,316 @@
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/babel-core/5.8.23/browser.min.js"></script>
+    <style>
+        .container {
+            padding: 5px;
+            border: 1px solid red;
+        }
+
+        .llist {
+            vertical-align: top;
+            padding: 5px;
+            margin-left: 5px;
+            margin-right: 5px;
+            border: 1px solid black;
+            margin-top: 20px;
+            width: 300px;
+            display: inline-block;
+
+        }
+        
+        .message {
+            padding: 5px;
+            margin: 10px auto;
+            width: 90%;
+            border: 1px solid #444;
+        }
+
+        input {
+            width: 240px;
+        }
+
+        .listname{
+            padding: 5px;
+            text-align: center;
+            font-weight: bold;
+        }
+
+        .messagetext {
+            margin-top: 11px;
+            margin-bottom: 11px;
+        }
+
+        .archived {
+            border: 1px solid grey;
+            color: grey;
+        }
+    </style>
+</head>
+<body>
+<div id="app"></div>
+    <script type="text/jsx">
+
+    var App = React.createClass({
+        getInitialState: function() {
+            return {  //This state can be changed to empty of course
+                    totalLists:2,
+                    totalMessages:4,
+                    inputListName:"",
+                    lists: [{listId: 1, listName: "first list1"},{listId: 2, listName: "second list2"}],
+                    allMessages: [  {messageId:1, belongsToList: 1, text: "testmessage1", IsArchived: false},
+                                    {messageId:2, belongsToList: 1, text: "testmessage2", IsArchived: false},
+                                    {messageId:3, belongsToList: 1, text: "testmessage3", IsArchived: false},
+                                    {messageId:4, belongsToList: 2, text: "testmessage4", IsArchived: false}]
+                };
+        },
+        eachList: function(list, i) {  
+        return (
+                <List key={i}
+                    index={i}
+                    listName={list.listName}
+                    listId={list.listId}
+                    allLists={this.state.lists}
+                    addMessageParent={this.addMessageParent}
+                    deleteMessageParent={this.deleteMessageParent}
+                    archiveMessageParent={this.archiveMessageParent}
+                    allMessages={this.state.allMessages}
+                >{list}</List>
+            );
+        },
+        createList: function(evt) {
+        if (this.state.inputListName=="") {
+            alert("Cannot add list with no name!")
+        } else {
+            var newlistId=this.state.totalLists
+            newlistId++; 
+            var newList = {
+                listId: newlistId,
+                listName: this.state.inputListName
+            };
+         
+            var lists=this.state.lists;
```

Comment:
You should never mutate state directly, as it can result a multiple sources of truth in you application, which leads to very hard to debug behavior. Use `this.setState` instead.

This will not create a new instance; it still points to the `this.state.lists`. Please use `Array.from` to create a new instance of the array. See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from